src/Utilities/SetHistory.js

```diff
@@ -1,14 +1,58 @@
 import queryString from 'query-string';
+import { ASC, DESC } from '../constants';
+
+export function setHistory(
+    history, systemIds = [], baselineIds = [], hspIds = [], referenceId, activeFactFilters = [], factFilter, stateFilters, factSort, stateSort
+) {
+    let nameFilters = [ ...activeFactFilters ];
```

Comment:
This could be simplifiedusing spread operator
```JS
let nameFilters = [ ...activeFactFilters, ...factFilter ? [factFilter] : [] ];
let filterState = [ ...stateFilters?.filter(({ selected }) => selected)?.map(({ filter }) => filter?.toLowerCase()) || [] ];
lest sort = [
  ...[ASC, DESC].includes(stateSort) ? [`${stateSort === DESC ? '-' : ''}state`] : [],
  ...[ASC, DESC].includes(factSort) ? [`${factSort === DESC ? '-' : ''}fact`] : [],
];
```